Build snippets during Analyze job by default

heaths opened the pull request:
Resolves #33877

heaths committed:
Build snippets during Analyze job by default

Resolves #33877

jsquire commented on sdk/iot/ci.yml:
```diff
@@ -20,8 +20,9 @@ extends:
   template: /eng/pipelines/templates/stages/archetype-sdk-client.yml
   parameters:
     ServiceDirectory: iot
+    BuildSnippets: true
```
If this is being done by default, do we need this here?   Maybe it was intended to be `false`?

heaths replied:
`true`, `false` - you say "toe-ma-toe" I say "toe-may-toe".

But, yes. :)

jsquire commented on sdk/dns/ci.mgmt.yml:
```diff
@@ -18,6 +18,7 @@ extends:
   template: /eng/pipelines/templates/stages/archetype-sdk-client.yml
   parameters:
     ServiceDirectory: dns
+    BuildSnippets: true
```
Since this is done by default, do we need this here?

annelo-msft reviewed (approved):
ACR LGTM, Thanks @heaths!

weshaggard reviewed (approved):
The change looks reasonable to me.

heaths committed:
Resolve PR feedback

heaths committed:
Don't build snippets for Core

Core has a few issues actually compiling with SNIPPET symbol because it
tries to elide some code for brevity. Fixing that changes the extracted
snippets and may make the samples more confusing so leaving as-is.

pallavit commented:
Do we want to create follow-up issues for the SDKs where we had to manually set this to false? This looks great, thank you for doing it.

heaths commented:
> Do we want to create follow-up issues for the SDKs where we had to manually set this to false? This looks great, thank you for doing it.

@pallavit already did. :) See https://github.com/Azure/azure-sdk-for-net/issues/20431. I updated the OP with individual tracking issues as tasks.
